test/libsolidity/SemanticTest.cpp

```diff
@@ -175,122 +177,109 @@ TestCase::TestResult SemanticTest::runTest(ostream& _stream, string const& _line
+					"The function " + test.call().signature + " is not known to the compiler"
+				);
+
+				output = callContractFunctionWithValueNoEncoding(
+					test.call().signature,
+					test.call().value.value,
+					test.call().arguments.rawBytes()
+				);
 			}
-			else if (!_compileViaYul && m_runWithYul)
-				AnsiColorized(_stream, _formatted, {BOLD, YELLOW})
-					<< _linePrefix << endl
-					<< _linePrefix << "Note that the test also has to pass via Yul." << endl;
-			return TestResult::Failure;
+
+			if (
+				m_transactionSuccessful == test.call().expectations.failure ||
+				output != test.call().expectations.rawBytes()
+			)
+				success = false;
+
+			test.setFailure(!m_transactionSuccessful);
+			test.setRawBytes(std::move(output));
+			test.setContractABI(m_compiler.contractABI(m_compiler.lastContractName()));
 		}
 	}
-	catch (WhiskersError const&)
-	{
-		// this is an error in Whiskers template, so should be thrown anyway
-		throw;
-	}
-	catch (YulException const&)
-	{
-		// this should be an error in yul compilation or translation
-		throw;
-	}
-	catch (boost::exception const&)
-	{
-		if (!_compileViaYul || m_runWithYul)
-			throw;
-	}
-	catch (std::exception const&)
+
+	if (success && !m_runWithYul && _compileViaYul)
 	{
-		if (!_compileViaYul || m_runWithYul)
-			throw;
+		m_compileViaYulCanBeSet = true;
+		AnsiColorized(_stream, _formatted, {BOLD, YELLOW}) <<
+			_linePrefix << endl <<
+			_linePrefix << "Test can pass via Yul and marked with compileViaYul: false." << endl;
+		return TestResult::Failure;
 	}
-	catch (...)
+
+	if (!success && (m_runWithYul || !_compileViaYul))
 	{
-		if (!_compileViaYul || m_runWithYul)
-			throw;
+		AnsiColorized(_stream, _formatted, {BOLD, CYAN}) << _linePrefix << "Expected result:" << endl;
+		for (auto const& test: m_tests)
+		{
+			ErrorReporter errorReporter;
+			_stream << test.format(errorReporter, _linePrefix, false, _formatted) << endl;
+			_stream << errorReporter.format(_linePrefix, _formatted);
+		}
+		_stream << endl;
+		AnsiColorized(_stream, _formatted, {BOLD, CYAN}) << _linePrefix << "Obtained result:" << endl;
+		for (auto const& test: m_tests)
+		{
+			ErrorReporter errorReporter;
+			_stream << test.format(errorReporter, _linePrefix, true, _formatted) << endl;
+			_stream << errorReporter.format(_linePrefix, _formatted);
+		}
+		AnsiColorized(_stream, _formatted, {BOLD, RED})
+			<< _linePrefix << endl
+			<< _linePrefix << "Attention: Updates on the test will apply the detected format displayed." << endl;
+		if (_compileViaYul && m_runWithoutYul)
+		{
+			_stream << _linePrefix << endl << _linePrefix;
+			AnsiColorized(_stream, _formatted, {RED_BACKGROUND}) << "Note that the test passed without Yul.";
+			_stream << endl;
+		}
+		else if (!_compileViaYul && m_runWithYul)
+			AnsiColorized(_stream, _formatted, {BOLD, YELLOW})
+				<< _linePrefix << endl
+				<< _linePrefix << "Note that the test also has to pass via Yul." << endl;
+		return TestResult::Failure;
 	}
```

Comment:
Are these not caught anymore?

Reply:
Yes, that is the point. This block is moved further inside and only catches unimplemented feature errors when forcibly compiling via yul.